test/bin/scenario.sh

```diff
@@ -1069,6 +1068,187 @@ EOF
+    shift
+
+    local full_vmname
+    full_vmname="$(full_vm_name "${vmname}")"
+    if [[ -n "${RUN_HOST_OVERRIDE}" ]]; then
+        vmname="${RUN_HOST_OVERRIDE}"
+        full_vmname="$(full_vm_name "${vmname}")"
+        ip=$(get_vm_ip "${full_vmname}")
+        set_vm_property "${vmname}" "ip" "${ip}"
+        set_vm_property "${vmname}" "ssh_port" "22"
+        set_vm_property "${vmname}" "api_port" "6443"
+        set_vm_property "${vmname}" "lb_port" "5678"
+    fi
+
+    local original_dir
+    original_dir="$(pwd)"
+
+    # Check/install oc
+    if ! command -v oc &> /dev/null; then
+        "${ROOTDIR}/scripts/fetch_tools.sh" "oc" || {
+            record_junit "${vmname}" "oc_installed" "FAILED"
+            exit 1
+        }
+    fi
+
+    local test_binary
+    if ! test_binary="$(build_ginkgo_binary "${vmname}")"; then
+        cd "${original_dir}"
+        exit 1
+    fi
+
+    # Set up test environment variables
+    local -r test_results_dir="${SCENARIO_INFO_DIR}/${SCENARIO}/gingko-results"
+    mkdir -p "${test_results_dir}"
+
+    # Set up kubeconfig for tests
+    local -r kubeconfig_file="${test_results_dir}/kubeconfig"
+    local -r vm_ip=$(get_vm_property "${vmname}" "ip")
+    local -r full_vmname="$(full_vm_name "${vmname}")"
+    local -r vm_hostname="${full_vmname/./-}"
+
+    # Get kubeconfig
+    run_command_on_vm "${vmname}" "sudo cp /var/lib/microshift/resources/kubeadmin/${vm_hostname}/kubeconfig /home/redhat/kubeconfig-${vm_hostname}" || {
+        error "Failed to copy kubeconfig from /var/lib/microshift/resources/kubeadmin/${vm_hostname}/kubeconfig"
+        record_junit "${vmname}" "setup_kubeconfig" "FAILED"
+        cd "${original_dir}"
+        exit 1
+    }
+    run_command_on_vm "${vmname}" "sudo chown redhat:redhat /home/redhat/kubeconfig-${vm_hostname}"
+
+    # Download kubeconfig to local file
+    copy_file_from_vm "${vmname}" "/home/redhat/kubeconfig-${vm_hostname}" "${kubeconfig_file}" || {
+        error "Failed to download kubeconfig from VM"
+        record_junit "${vmname}" "setup_kubeconfig" "FAILED"
+        cd "${original_dir}"
+        exit 1
+    }
+    record_junit "${vmname}" "setup_kubeconfig" "OK"
+
+    export KUBECONFIG="${kubeconfig_file}"
+
+    # Run the Gingko tests with MicroShift filter
+    echo "Running Gingko tests with MicroShift filter..."
+    echo "Executing: ${test_binary} run all --dry-run | grep \"MicroShift\" | ${test_binary} run -f - --timeout 60m"
+
+    # Run the tests and capture output
+    if ! ${test_binary} run all --dry-run | grep "MicroShift" | ${test_binary} run -f - --timeout 60m 2>&1 | tee "${test_results_dir}/test-output.log"; then
+        echo "Some Gingko tests may have failed. Check results for details."
+        record_junit "${vmname}" "run_gingko_tests" "FAILED"
+        cd "${original_dir}"
+        return 1
+    fi
+
+    record_junit "${vmname}" "run_gingko_tests" "OK"
+
+    # Restore original directory
+    cd "${original_dir}"
```

Comment:
This is usually done with `popd &>/dev/null`

Reply:
Done.